cbmc/proofs/IotMqtt_UnsubscribeAsync/Makefile

```diff
@@ -0,0 +1,116 @@
+ABSTRACTIONS += --remove-function-body _matchEndWildcards
+ABSTRACTIONS += --remove-function-body _matchWildcards
+ABSTRACTIONS += --remove-function-body _mqttOperation_match
+ABSTRACTIONS += --remove-function-body _mqttSubscription_setUnsubscribe
+ABSTRACTIONS += --remove-function-body _packetMatch
+ABSTRACTIONS += --remove-function-body _topicFilterMatch
+ABSTRACTIONS += --remove-function-body _validateQos
+
+# Functions manually determined to be unreachable in their calling context
+# See README.md for explanations of unreachability
+ABSTRACTIONS += --remove-function-body _checkRetryLimit
+ABSTRACTIONS += --remove-function-body _scheduleNextRetry
+ABSTRACTIONS += --remove-function-body _IotMqtt_RemoveSubscriptionByPacket
+
+################################################################
+
+OBJS += $(ENTRY)_harness.goto
+OBJS += $(MQTT)/cbmc/proofs/mqtt_state.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_api.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_helper.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_operation.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_serialize.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_subscription.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_validate.goto
+
+DEF += -DCBMC_CREATEOPERATION_INITIALIZE_CANT_FAIL=1
+DEF += -DCBMC_CREATEOPERATION_UNSUCCESSFUL_STATUS_MEANS_NULL_POINTER=1
+DEF += -DCBMC_FINDFIRSTMATCH_MATCH_IS_NONNULL=1
+DEF += -DCBMC_STUB_REMOVEALLMATCHES=1
+
+# allow mqtt mallocs to fail for coverage
+DEF += -include mqtt_state.h
+DEF += -DIotMqtt_MallocMessage=malloc_can_fail
+DEF += -DIotMqtt_MallocOperation=malloc_can_fail
+
+################################################################
+# Loop unwinding
+
+LOOP += IotListDouble_FindFirstMatch.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link1.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link2.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += IotListDouble_FindFirstMatch\$$link5.0:$(SUBSCRIPTION_COUNT_MAX)
+
+LOOP += IotListDouble_RemoveAllMatches.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link1.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link2.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link3.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link4.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link5.0:$(SUBSCRIPTION_LIST_MAX)
+LOOP += IotListDouble_RemoveAllMatches\$$link6.0:$(SUBSCRIPTION_LIST_MAX)
+
+LOOP += _IotMqtt_RemoveSubscriptionByTopicFilter.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribe.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SerializeUnsubscribeCommon.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_SubscriptionPacketSize.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _IotMqtt_ValidateSubscriptionList.0:$(SUBSCRIPTION_COUNT_MAX)
+LOOP += _encodeRemainingLength.0:2
```

Comment:
Add
```Makefile
# One more than actual length of payload
PAYLOAD_LENGTH_MAX=6
DEF += -DPAYLOAD_LENGTH_MAX=$(PAYLOAD_LENGTH_MAX)
```
And replace 2 by `$(PAYLOAD_LENGTH_MAX)`

Reply:
This really is a constant.  This loops encodes a length into a sequence of bytes, it does not scan a buffer of a given length.